packages/miniflare/src/plugins/kv/assets.ts

```diff
@@ -18,68 +19,53 @@ export function isWorkersWithAssets(
 	return options.assetsPath !== undefined;
 }
 
-async function* listKeysInDirectoryInner(
-	rootPath: string,
-	currentPath: string
-): AsyncGenerator<string> {
-	const fileEntries = await fs.readdir(currentPath, { withFileTypes: true });
-	for (const fileEntry of fileEntries) {
-		const filePath = path.posix.join(currentPath, fileEntry.name);
-		if (fileEntry.isDirectory()) {
-			yield* listKeysInDirectoryInner(rootPath, filePath);
-		} else {
-			// Get key name by removing root directory & path separator
-			// (assumes `rootPath` is fully-resolved)
-			yield filePath.substring(rootPath.length + 1);
-		}
-	}
-}
-
-function listKeysInDirectory(rootPath: string): AsyncGenerator<string> {
-	rootPath = path.resolve(rootPath);
-	return listKeysInDirectoryInner(rootPath, rootPath);
-}
-
 const SERVICE_NAMESPACE_ASSET = `${KV_PLUGIN_NAME}:asset`;
 
-async function buildStaticContentManifest(assetsPath: string) {
-	// Build __STATIC_CONTENT_MANIFEST contents
-	const staticContentManifest: Record<string, string> = {};
-	for await (const key of listKeysInDirectory(assetsPath)) {
-		staticContentManifest[key] = encodeAssetsKey(key);
-	}
-	return staticContentManifest;
+export function buildAssetsManifest(): Uint8Array {
+	const buffer = new ArrayBuffer(20);
+	const assetManifest = new Uint8Array(buffer); // [0, 0, 0, ..., 0, 0]
+	assetManifest.set([1], 0); // [1, 0, 0, ..., 0, 0]
+
+	return assetManifest;
 }
 
 export async function getAssetsBindings(
 	options: AssetsOptions
 ): Promise<Worker_Binding[]> {
-	const __STATIC_CONTENT_MANIFEST = await buildStaticContentManifest(
-		options.assetsPath
+	const assetsBindings = getAssetsBindingsNames(
+		options?.assetsKVBindingName,
+		options?.assetsManifestKVBindingName
 	);
 
+	const assetsManifest = buildAssetsManifest();
+
 	return [
 		{
-			name: AssetsBindings.KV_NAMESPACE_ASSET,
+			// this is the binding to the KV namespace that the assets are in.
+			name: assetsBindings.ASSETS_KV_NAMESPACE,
 			kvNamespace: { name: SERVICE_NAMESPACE_ASSET },
 		},
 		{
-			name: AssetsBindings.JSON_ASSET_MANIFEST,
-			json: JSON.stringify(__STATIC_CONTENT_MANIFEST),
+			// this is the binding to an ArrayBuffer containing the binary-encoded
+			// assets manifest.
+			name: assetsBindings.ASSETS_MANIFEST,
+			data: assetsManifest,
 		},
 	];
 }
 
 export async function getAssetsNodeBindings(
 	options: AssetsOptions
 ): Promise<Record<string, unknown>> {
-	const __STATIC_CONTENT_MANIFEST = await buildStaticContentManifest(
-		options.assetsPath
+	const assetsManifest = await buildAssetsManifest();
```

Comment:
I'm not 100% sure if `getAssetsNodeBindings` is relevant for our work :/

Reply:
"Node bindings" are to support `getPlatformProxy()` which is used by framework dev tools, such as Next.js.